pug/index.d.ts

```diff
@@ -16,35 +16,122 @@ declare module 'pug' {
+    /**
+     * Compile a Pug template to a function which can be rendered multiple times with different locals.
+     */
+    export function compile(template: string, options?: Options): compileTemplate;
+
+    /**
+     * Compile a Pug template from a file to a function which can be rendered multiple times with different locals.
+     */
+    export function compileFile(path: string, options?: Options): compileTemplate;
+
+    /**
+     * Compile a Pug template to a string of JavaScript that can be used client side along with the Pug runtime.
+     */
+    export function compileClient(template: string, options?: Options): string;
+
+    /**
+     * Compile a Pug template to an object of the form:
+     * {
+     *   'body': 'function (locals) {...}',
+     *   'dependencies': ['filename.pug']
+     * }
+     * that can be used client side along with the Pug runtime.
+     * You should only use this method if you need dependencies to implement something like watching for changes to the Pug files.
+     */
+    export function compileClientWithDependenciesTracked(template: string, options?: Options): {
+        body: string;
         dependencies: string[];
     };
-    export function compileFileClient(path: string, options?: Options): ClientFunctionString;
-    export function render(source: string, options?: Options, callback?: (err: Error, html: string) => void): string;
-    export function renderFile(path: string, options?: Options, callback?: (err: Error, html: string) => void): string;
 
-    // else
-    export type ClientFunctionString = string;  // ex: 'function (locals) {...}'
-    export type compileTemplate = (locals?: any) => string;
+    /**
+     * Compile a Pug template file to a string of JavaScript that can be used client side along with the Pug runtime.
+     */
+    export function compileFileClient(path: string, options?: Options): string;
+
+    /**
+     * Compile a Pug template and render it without locals to html string.
+     */
+    export function render(template: string): string;
+    /**
+     * Compile a Pug template and render it with locals to html string.
+     * @param {(Options & LocalsObject)} options Pug Options and rendering locals
+     */
+    export function render(template: string, options: Options & LocalsObject): string;
+    /**
+     * Compile a Pug template and render it without locals to html string.
+     */
+    export function render(template: string, callback: (err: Error | null, res: string) => void): void;
+    /**
+     * Compile a Pug template and render it with locals to html string.
+     * @param {(Options & LocalsObject)} options Pug Options and rendering locals
+     * @param {((err: Error | void, res: string) => void)} callback Node.js-style callback receiving the rendered results. This callback is called synchronously.
+     */
+    export function render(template: string, options: Options & LocalsObject, callback: (err: Error | null, res: string) => void): void;
```

Comment:
Oh, I see. It's because I modified the original typings from types/npm-pug#5 and now I wanted to move them to `@types` on npm and remove `typings` dependency from my project's package.json.
Fixed in e573985.